test/gurka/test_conditional_restrictions.cc

```diff
@@ -0,0 +1,144 @@
+#include "gurka.h"
+#include <gtest/gtest.h>
+
+#if !defined(VALHALLA_SOURCE_DIR)
+#define VALHALLA_SOURCE_DIR
+#endif
+
+using namespace valhalla;
+
+class ConditionalRestrictions : public ::testing::Test {
+protected:
+  static gurka::map map;
+
+  static void SetUpTestSuite() {
+    constexpr double gridsize = 100;
+
+    const std::string ascii_map = R"(
+         A--B
+         |  \
+         D--C)";
+
+    const gurka::ways ways = {
+        {"AD", {{"highway", "service"},{"motorcar", "no"},{"bicycle", "yes"},{"foot", "yes"},
+          {"bicycle:conditional",
+           "no @ (Mar 00:00-07:00;Mar 17:30-24:00;Apr 00:00-07:00;Apr 19:00-24:00;Aug 00:00-07:00;Aug 19:00-24:00;Sep 00:00-07:00;Sep 18:00-24:00;Oct 00:00-07:00;Oct 17:00-24:00;Jan-Feb 00:00-07:30;Jan-Feb 17:00-24:00;May-Jul 00:00-07:00;May-Jul 20:00-24:00)"},
+          {"foot:conditional",
+           "yes @ (Mar 00:00-07:00;Mar 17:30-24:00;Apr 00:00-07:00;Apr 19:00-24:00;Aug 00:00-07:00;Aug 19:00-24:00;Sep 00:00-07:00;Sep 18:00-24:00;Oct 00:00-07:00;Oct 17:00-24:00;Jan-Feb 00:00-07:30;Jan-Feb 17:00-24:00;May-Jul 00:00-07:00;May-Jul 20:00-24:00)"}
+        }},
+        {"AB",
+         {{"highway", "service"},
+          {"motorcar:conditional",
+           "yes @ (Mar 00:00-07:00;Mar 17:30-24:00;Apr 00:00-07:00;Apr 19:00-24:00;Aug 00:00-07:00;Aug 19:00-24:00;Sep 00:00-07:00;Sep 18:00-24:00;Oct 00:00-07:00;Oct 17:00-24:00;Jan-Feb 00:00-07:30;Jan-Feb 17:00-24:00;May-Jul 00:00-07:00;May-Jul 20:00-24:00)"},
```

Comment:
Can the time string (`(Mar 00:00-07:00...)`) be made into variable in the test? It looks like all the strings are the same and we can just use a variable instead.

Reply:
To expand on that - does the time condition need to be this long at all?  Can we test the intent of the change with just a couple of time periods?